### PR TITLE
Release 02012021

### PR DESCRIPTION
**Pull request by ssl1321ois**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -64,6 +64,8 @@ class App extends Component {
     localStorage.setItem('endDate', endDate)
   }
 
+
```
Comment (ssl1321ois) [inline src/App.js:67]: delete

File: src/actions/index.jsx [hunk 2]
```diff
@@ -29,12 +29,15 @@ export function addActivityType(userID, token, activityTypeName, isEnable, isPri
     }
 }
 
-export function removeActivityType(userID, token, activityTypeName) {
+export function removeActivityType(userID, token, targetActivityTypeName, activityTypeName, isEnable, isPrivate) {
     return {
         type: "REMOVE_ACTIVITY_TYPE",
         userID: userID,
         token: token,
+        targetActivityTypeName: targetActivityTypeName,
```
Comment (ssl1321ois) [inline src/actions/index.jsx:37]: reverse all the changes

File: src/middleware/index.jsx [hunk 3]
```diff
@@ -106,7 +107,10 @@ const myMiddleware = store => next => action => {
         const headers = getHeaders(action.token)
         const body = {
             userID: action.userID,
-            activityTypeName: action.activityTypeName
+            targetActivityTypeName: action.targetActivityTypeName,
```
Comment (ssl1321ois) [inline src/middleware/index.jsx:110]: reverse changes

File: src/middleware/index.jsx [hunk 4]
```diff
@@ -140,6 +144,7 @@ const myMiddleware = store => next => action => {
         const body = getBody(action.userID)
         body.startDate = moment(localStorage.getItem("startDate")).format("YYYY/MM/DD")
         body.endDate = moment(localStorage.getItem("endDate")).format("YYYY/MM/DD")
+        console.log(body.startDate);
```
Comment (ssl1321ois) [inline src/middleware/index.jsx:147]: delete

File: src/views/Activity.jsx [hunk 5]
```diff
@@ -106,7 +106,10 @@ class Activity extends Component {
                 this.props.removeActivityType(
                   localStorage.getItem("uid"),
                   null,
-                  oldData.name
+                  oldData.name,
```
Comment (ssl1321ois) [inline src/views/Activity.jsx:109]: reverse changes

File: src/views/Activity.jsx [hunk 6]
```diff
@@ -130,8 +133,8 @@ function mapDispatchToProps(dispatch) {
       dispatch(editActivityType(userID, token, targetActivityTypeName, activityTypeName, isEnable, isPrivate)),
     addActivityType: (userID, token, activityTypeName, isEnable, isPrivate) =>
       dispatch(addActivityType(userID, token, activityTypeName, isEnable, isPrivate)),
-    removeActivityType: (userID, token, activityTypeName) =>
-      dispatch(removeActivityType(userID, token, activityTypeName))
+    removeActivityType: (userID, token, targetActivityTypeName, activityTypeName, isEnable, isPrivate) =>
+      dispatch(removeActivityType(userID, token, targetActivityTypeName, activityTypeName, isEnable, isPrivate))
```
Comment (ssl1321ois) [inline src/views/Activity.jsx:137]: reverse changes

File: src/views/Appbar.jsx [hunk 7]
```diff
@@ -114,3 +119,5 @@ function mapStateToProps(state) {
 }
 
 export default connect(mapStateToProps, null)(Appbar)
+
+
```
Comment (ssl1321ois) [inline src/views/Appbar.jsx:123]: delete

File: src/views/Board.jsx [hunk 8]
```diff
@@ -3,40 +3,223 @@ import { withRouter } from "react-router-dom";
...
+        this.setState({ select: false})
+        break
+      }
+    }
+    this.setState({ activityTypeList: this.state.activityTypeList});
+  }
+
+  handleSelectAll(event) {
+    if( this.state.select === false) {
+      console.log("false")
```
Comment (ssl1321ois) [inline src/views/Board.jsx:104]: delete

File: src/views/Board.jsx [hunk 9]
```diff
@@ -3,40 +3,223 @@ import { withRouter } from "react-router-dom";
...
+
+  handleSelectAll(event) {
+    if( this.state.select === false) {
+      console.log("false")  
+      for(const each of this.state.activityTypeList) {
+          each.checked = true;
+        }
+        this.state.select= true;
+      }else {
+        console.log("true")
```
Comment (ssl1321ois) [inline src/views/Board.jsx:110]: delete

File: src/views/Board.jsx [hunk 10]
```diff
@@ -3,40 +3,223 @@ import { withRouter } from "react-router-dom";
...
+    for(const each of this.state.activityTypeList){
+      if( each.checked === true){
+        this.state.filterList.push(each.name)
+      }
+    }
+    this.props.updateDashBoard(
+      localStorage.getItem("uid"),
+      null,
+      this.state.filterList)  
+    console.log( this.state.activityTypeList)
```
Comment (ssl1321ois) [inline src/views/Board.jsx:152]: delete

File: src/views/Board.jsx [hunk 11]
```diff
@@ -3,40 +3,223 @@ import { withRouter } from "react-router-dom";
...
+  }
+
   render() {
+    const { classes } = this.props;
+    // this.initialize()
```
Comment (ssl1321ois) [inline src/views/Board.jsx:157]: delete

**Review by Nightlord851108 — APPROVED** [review]
good

**Review by Nightlord851108 — COMMENTED** [review]
Good

**Review by Nightlord851108 — APPROVED** [review]
Good

**Pull request merged**